File: go/main/3-longest-substring-without-repeating-characters/3_test.go
package __longest_substring_without_repeating_characters

import "testing"

type _data struct {
	Str string
	Len int
}

func TestLengthOfLongestSubstring(t *testing.T) {
	data := []_data{
		_data{
			Str: "bbbbb",
			Len: 1,
		},
		_data{
			Str: "abcabcbb",
			Len: 3,
		},
		_data{
			Str: "pwwkew",
			Len: 3,
		},
		_data{
			Str: "dvdf",
			Len: 3,
		},
	}

	for i := range data {
		if l := lengthOfLongestSubstring(data[i].Str); data[i].Len != l {
			t.Errorf("(%s) (%d): %d != %d", data[i].Str, data[i].Len, l, data[i].Len)
		}
	}
}

File: c/src/main/3-longest-substring-without-repeating-characters/3_test.c
#include <criterion/criterion.h>
#include "3.h"

Test(lengthOfLongestSubstring, a001) {
    char *str = "abcabcbb";
    int expected = 3;

    cr_assert_eq(expected, lengthOfLongestSubstring(str));
}

Test(lengthOfLongestSubstring, a002) {
    char *str = "bbbbb";
    int expected = 1;

    cr_assert_eq(expected, lengthOfLongestSubstring(str));
}

Test(lengthOfLongestSubstring, a003) {
    char *str = "pwwkew";
    int expected = 3;

    cr_assert_eq(expected, lengthOfLongestSubstring(str));
}

Test(lengthOfLongestSubstring, a004) {
    char *str = "";
    int expected = 0;

    cr_assert_eq(expected, lengthOfLongestSubstring(str));
}

Test(lengthOfLongestSubstring, a005) {
    char *str = " ";
    int expected = 1;

    cr_assert_eq(expected, lengthOfLongestSubstring(str));
}

Test(lengthOfLongestSubstring, a006) {
    char *str = "tmmzuxt";
    int expected = 5;

    cr_assert_eq(expected, lengthOfLongestSubstring(str));
}

Test(lengthOfLongestSubstring, a007) {
    char *str = "dvdf";
    int expected = 3;

    cr_assert_eq(expected, lengthOfLongestSubstring(str));
}

Test(lengthOfLongestSubstring, a008) {
    char *str = "ohvhjdml";
    int expected = 6;

    cr_assert_eq(expected, lengthOfLongestSubstring(str));
}